File: bank-app/app/src/main/java/com/android/bankapp/statements/interactor/StatementInteractor.java
package com.android.bankapp.statements.interactor;

import com.android.bankapp.statements.presenter.StatementsPresenter;

public class StatementInteractor implements StatementInteractorInput {

    public StatementsPresenter output;

    @Override
    public void loadData() {
        output.loadData();
    }
}

File: BankApp/app/src/main/java/br/com/santander/android/bank/statement/domain/interactor/StatementInteractor.kt
package br.com.santander.android.bank.statement.domain.interactor

import br.com.santander.android.bank.statement.domain.model.Statements
import br.com.santander.android.bank.statement.repository.StatementRepository
import io.reactivex.Observable

internal class StatementInteractor(private val statementRepository: StatementRepository) {

    fun getStatements(userId: Int): Observable<Statements> {
        return statementRepository.getStatements(userId)
    }
}